### DISCUSSION TITLE
When I perform adversarial training, I find that the output of BERT is always NAN.

### DISCUSSION THREAD

**Discussion by Struggle-Forever** [Lightning-AI/pytorch-lightning#13546] [code help: NLP / ASR / TTS] [answered]
- [ ] When I perform adversarial training, I find that the output of BERT is always NAN.
- [ ] 
![微信截图_20220705202316](https://user-images.githubusercontent.com/29114869/177455365-d167b314-5776-496a-8414-8485a40ac04a.png)

**Comment by akihironitta** [comment, marked as answer]
As far as I know, training with AMP (`precision=16`) can sometimes be unstable and lead to nan as you report.

**Reply by Struggle-Forever** [reply]
trainer = Trainer(devices="auto", accelerator="auto", logger=False, multiple_trainloader_mode='max_size_cycle',
                          terminate_on_nan=True,
                          # logger=tb_logger, log_every_n_steps=1,
                          # flush_logs_every_n_steps=5,
                          max_epochs=args.epochs)


I find that sometimes I can train normally and sometimes it is NAN

**Reply by Struggle-Forever** [reply]
I don't know what happened, but when I chose a new GPU, the NAN problem was solved. I hope this method can help everyone.
